extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -374,30 +447,38 @@ function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
                 StudyInstanceUID: naturalized.StudyInstanceUID,
               };
               // Todo: this needs to be from wado dicom web client
-              return qidoDicomWebClient.retrieveBulkData(options).then(val => {
-                // There are DICOM PDF cases where the first ArrayBuffer in the array is
-                // the bulk data and DICOM video cases where the second ArrayBuffer is
-                // the bulk data. Here we play it safe and do a find.
-                const ret =
-                  (val instanceof Array && val.find(arrayBuffer => arrayBuffer?.byteLength)) ||
-                  undefined;
-                value.Value = ret;
-                return ret;
-              });
+              return clientManager
+                .getQidoClient()
+                .retrieveBulkData(options)
+                .then(val => {
+                  // There are DICOM PDF cases where the first ArrayBuffer in the array is
+                  // the bulk data and DICOM video cases where the second ArrayBuffer is
+                  // the bulk data. Here we play it safe and do a find.
+                  const ret =
+                    (val instanceof Array && val.find(arrayBuffer => arrayBuffer?.byteLength)) ||
+                    undefined;
+                  value.Value = ret;
+                  return ret;
+                });
             };
           }
         });
         return naturalized;
       };
 
       // Async load series, store as retrieved
-      function storeInstances(instances) {
+      function storeInstances(instances, clientName) {
         const naturalizedInstances = instances.map(addRetrieveBulkData);
 
         // Adding instanceMetadata to OHIF MetadataProvider
+        const client = clientManager.getClient(clientName);
```

Comment:
you should just get the client above, no one will switch client during the call, no?